core/src/credential.rs

```diff
@@ -0,0 +1,97 @@
+
+#![allow(unused_variables)]
+#![allow(dead_code)]
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+type VCContext = [String];
+
+pub(crate) const JSON_LD_CONTEXT:  VCContext = ["https://www.w3.org/2018/credentials/v1", "https://www.w3.org/2018/credentials/examples/v1"];
+pub(crate) const CRED_TYPE_PERMANENT_RESIDENT_CARD: str = "PermanentResidentCard";
+pub(crate) const CRED_TYPE_BANK_CARD: str = "BankCard";
+
+type CredentialSubject = serde_json::Value;
+
+enum CredType {
+    PermanentResidentCard(String),
+    BankCard(String)
+}
+
+pub trait VCCredential {
+    fn getCredType (&self) -> String;
+    fn getSubject (&self) -> String;
+    fn getContext (&self) -> [String];
+}
+
+impl VCCredential for VerifiableCredential<'_> {
+    fn getCredType (&self) -> String {
+        self.cred_type = ""
+    }
+    fn getSubject (&self) -> String {
+        self.subject = ""
+    }
+    fn getContext (&self) ->[String] {
+        return crate::credential::JSON_LD_CONTEXT
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct VerifiableCredential <'a> {
+    #[serde(rename = "@context")]
+	context:  VCContext,
+	id: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    cred_type:  Option<CredType>,
```

Comment:
I don't think we should have an enum here, since users can override the cred type we don't have those kinds of guarantees. Also why is this an `Option<T>`?

Reply:
The Option<T> here was to handle the case where there is no cred type, but from our morning conversation i realize that it won't be the case anyway, we are returning error if cred_type is redefined and empty.